cellarium/ml/core/module.py

```diff
@@ -416,3 +402,63 @@ def on_save_checkpoint(self, checkpoint: dict[str, Any]) -> None:
                 checkpoint["loops"]["fit_loop"]["epoch_progress"]["total"]["completed"] += 1
                 checkpoint["loops"]["fit_loop"]["epoch_progress"]["current"]["completed"] += 1
                 checkpoint["CellariumAnnDataDataModule"]["epoch"] += 1
+
+
+# define a decorator to dynamically add methods
+def add_delegate_method(method_name):
+    def delegate_method(self, *args, **kwargs):
+        method = getattr(self.model, method_name, None)
+        if callable(method):
+            method(self.trainer)
```

Comment:
Pass `*args` and `**kwargs` here as well.

Reply:
So I did try this initially:
```python
method(self.trainer, *args, **kwargs)
```
but I got unexpected test failures...
then I realized that the explicitly implemented ones only used the `self.trainer` arg.

Reply:
Should we be passing args and kwargs?

Reply:
I admit I did not understand the test failures.  It seemed like something funny was happening during "predict".  I gave up on args and kwargs for that reason, thinking it must be wrong.  But I never understood the test failure.